File: src/main/java/com/tribalscale/test/mapper/MoneyTransferMapper.java
package com.tribalscale.test.mapper;

import com.tribalscale.test.domain.MoneyTransfer;
import com.tribalscale.test.domain.payload.MoneyTransferRequestPayload;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface MoneyTransferMapper {
    MoneyTransfer payloadToEntity(MoneyTransferRequestPayload payload);

    MoneyTransferRequestPayload entityToPayload(MoneyTransfer entity);
}

File: src/main/java/com/tribalscale/test/mapper/MoneyTransferMapper.kt
package com.tribalscale.test.mapper

import com.tribalscale.test.domain.payload.MoneyTransferRequestPayload
import com.tribalscale.test.domain.MoneyTransfer
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface MoneyTransferMapper {
    fun payloadToEntity(payload: MoneyTransferRequestPayload): MoneyTransfer
    fun entityToPayload(entity: MoneyTransfer): MoneyTransferRequestPayload
}